### PR TITLE
test/fuzz: add crypto/merkle.FuzzProofsFromByteSlices

### PR DESCRIPTION
**Pull request by odeke-em**
Adds a fuzzer to exorcise merkle.ProofsFromByteSlices.

Replaces PR https://github.com/tendermint/tendermint/pull/8470 which got spuriously closed by a Github actions bot while I was out of the office.

### CODE REVIEW

File: test/fuzz/tests/crypto_merkle_test.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+		total := 100
+		seed := make([][]byte, total)
+		for j := 0; j < total; j++ {
+			seed[j] = tmrand.Bytes(tmcrypto.HashSize)
+		}
+		seeds = append(seeds, bytes.Join(seed, fakeSeparator))
+	}
+
+	for _, seed := range seeds {
+		f.Add(seed)
```
Comment (elias-orijtech) [inline test/fuzz/tests/crypto_merkle_test.go:33]: github.com/orijtech/otils/corpus2ossfuzz only supports corpus files from the testdata/fuzz/FuzzXXX directories. Otherwise we can leave this as is and https://github.com/google/oss-fuzz/issues/7600 will make this moot.

**Review by elias-orijtech — COMMENTED** [review]
If you intend to run this fuzzer on OSS-Fuzz please add a line to `/test/fuzz/oss-fuzz-build.sh`. Also see the note about `testing.F.Add`.

**Review by tychoish — CHANGES_REQUESTED** [review]
It might be good if the context for this to be more apparent in the test. From reading the code I’m not sure what this test is supposed to be asserting (e.g. there’s not obvious failure case.)

Also want to second the comment that I’m not super sure that this test is ever going to execute, and it might be good to look into that before we merge this

File: test/fuzz/tests/crypto_merkle_test.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	}
+
+	f.Fuzz(func(t *testing.T, joinedInput []byte) {
```
[the remark this answers is not in the document]
Comment (elias-orijtech) [reply test/fuzz/tests/crypto_merkle_test.go:36]: Usually, fuzzing is used to tease out corner cases where a parser/decoder panics because of unexpected input. So while you can use `f.Errorf` or similar to flag a failing fuzz test, this is the usual pattern.

Besides, OSS-Fuzz [doesn't support](https://github.com/golang/go/issues/50192) native Go 1.18 fuzzing yet and [will ignore](https://github.com/AdamKorcz/go-118-fuzz-build/blob/7bed8f92ea820f668efab099ec774d548f1873a8/utils/test_helpers.go#L19) `f.Errorf` and friends.

**Comment by elias-orijtech** [comment]
> Also want to second the comment that I’m not super sure that this test is ever going to execute, and it might be good to look into that before we merge this

`go test` won't fuzz without asking it to, but it will test the seed corpus. I assume @odeke-em will add a line to `oss-fuzz-build-sh` to add the test to the tendermint OSS-Fuzz project which will take care of fuzzing for new failing cases.

**Comment by cmwaters** [comment]
@odeke-em - Is this PR ready for another round of reviews? I'm not quite sure what state it's in and just want to follow up.
